server/routes/inPrisonCoursesAndQualifications/inPrisonCoursesAndQualificationsView.ts

```diff
@@ -1,13 +1,33 @@
-import type { PrisonerSummary } from 'viewModels'
+import type { InPrisonCourse, InPrisonCourseRecords, PrisonerSummary } from 'viewModels'
+import dateComparator from '../dateComparator'
 
 export default class InPrisonCoursesAndQualificationsView {
-  constructor(private readonly prisonerSummary: PrisonerSummary) {}
+  constructor(
+    private readonly prisonerSummary: PrisonerSummary,
+    private readonly inPrisonCourses: InPrisonCourseRecords,
+  ) {}
 
   get renderArgs(): {
     prisonerSummary: PrisonerSummary
+    problemRetrievingData: boolean
+    completedCourses: Array<InPrisonCourse>
+    inProgressCourses: Array<InPrisonCourse>
+    withdrawnCourses: Array<InPrisonCourse>
   } {
     return {
       prisonerSummary: this.prisonerSummary,
+      problemRetrievingData: this.inPrisonCourses.problemRetrievingData,
+      completedCourses: [...(this.inPrisonCourses.coursesByStatus?.COMPLETED || [])].sort((left, right) =>
+        dateComparator(left.courseCompletionDate, right.courseCompletionDate, 'DESC'),
+      ),
+      inProgressCourses: [...(this.inPrisonCourses.coursesByStatus?.IN_PROGRESS || [])].sort((left, right) =>
+        dateComparator(left.courseStartDate, right.courseStartDate, 'DESC'),
+      ),
+      withdrawnCourses: [
+        ...(this.inPrisonCourses.coursesByStatus?.WITHDRAWN || []).concat(
+          this.inPrisonCourses.coursesByStatus?.TEMPORARILY_WITHDRAWN || [],
+        ),
+      ].sort((left, right) => dateComparator(left.courseStartDate, right.courseStartDate, 'DESC')),
```

Comment:
We pass the entire `InPrisonCourseRecords` object into the view class, and then lets it's `renderArgs` method work out the bits that the view needs. IE. separating the `coursesByStatus` records into separate properties, sorting them as the view needs them, and in the case of withdrawn courses combining 2 arrays (withdrawn and temporarily withdrawn)
It's better here than trying to do in in the nunjucks template 👍